shuffle/move/examples/main/mod.ts

```diff
@@ -44,123 +39,93 @@ export async function printWelcome() {
-    ),
-  );
-
-  const payload = codegen.Stdlib.encodeInitializeNftCollectionScriptFunction(
-    testNftType,
-  );
-  return await DiemHelpers.buildAndSubmitTransaction(
-    defaultUserContext.address,
-    await devapi.sequenceNumber(),
-    await defaultUserContext.readPrivateKey(),
-    payload,
+  return await invokeScriptFunction(
+    "Message::set_message",
+    [],
+    [message],
+    sender,
+    moduleAddress
   );
 }
 
 // ScriptFunction example; creation of NFT.
-// See main/source/TestNFT.move
+// See main/source/nft/TestNFT.move
 export async function createTestNFTScriptFunction(
-  userContext: UserContext,
   contentUri: string,
+  sender?: UserContext,
+  moduleAddress?: string,
 ) {
-  const scriptFunction: string = userContext.address +
-    "::TestNFT::create_nft";
-  const typeArguments: string[] = [];
-  const args: any[] = [contentUri];
-  return await DiemHelpers.invokeScriptFunctionForContext(
-    userContext,
-    scriptFunction,
-    typeArguments,
-    args,
+  return await invokeScriptFunction(
+    "TestNFT::create_nft",
+    [],
+    [contentUri],
+    sender,
+    moduleAddress
   );
 }
 
 // ScriptFunction example; creation of NFT.
 // See main/source/TestNFT.move
 export async function transferNFTScriptFunction(
-  userContext: UserContext,
   to: string,
   creator: string,
   creationNum: StringLiteral,
+  sender?: UserContext,
+  moduleAddress?: string,
 ) {
-  const scriptFunction: string = userContext.address +
-    "::NFTStandard::transfer";
-  const typeArgument = userContext.address + "::TestNFT::TestNFT";
-  const typeArguments: string[] = [typeArgument];
-  const args: any[] = [to, creator, creationNum];
-  return await DiemHelpers.invokeScriptFunctionForContext(
-    userContext,
-    scriptFunction,
-    typeArguments,
-    args,
+  moduleAddress = moduleAddress || defaultUserContext.address;
+
+  return await invokeScriptFunction(
+    "NFTStandard::transfer",
+    [`${moduleAddress}::TestNFT::TestNFT`],
+    [to, creator, creationNum],
+    sender,
+    moduleAddress
   );
 }
 
-export async function initializeNFTScriptFunction(userContext: UserContext) {
-  const scriptFunction: string = defaultUserContext.address +
-    "::NFTStandard::initialize_nft_collection";
+// Initialize NFT
+// See main/source/nft/NFTStandard.move
+// See main/source/nft/TestNFT.move
+export async function initializeNFTScriptFunction(
+  sender?: UserContext,
+  moduleAddress?: string,
+) {
+  moduleAddress = moduleAddress || defaultUserContext.address;
+
+  return await invokeScriptFunction(
+    "NFTStandard::initialize_nft_collection",
+    [`${moduleAddress}::TestNFT::TestNFT`],
+    [],
+    sender,
+    moduleAddress
+  );
+}
 
-  const typeArgument = defaultUserContext.address + "::TestNFT::TestNFT";
-  const typeArguments: string[] = [typeArgument];
+export async function invokeScriptFunction(
+  funcName: string,
+  typeArgs: string[],
+  args: any[],
+  sender?: UserContext,
+  moduleAddress?: string,
```

Comment:
I'm still conflicted about this method. I imagine a scenario where someone wants to call a script function that's not theirs in console, like "0x1::NFT::transfer". Now they have two paths:

```
> await main.invokeScriptFunction("NFT::transfer,[],[mv.Raw("0x123")],defaultUserContext, "0x1");

// or

> await helpers.invokeScriptFunction("0x1::NFT::transfer, [], [mv.Raw("0x123")]);
```

Can we not export this method and only use it internally? Why export two `invokeScriptFunction`s?

Reply:
as you wish